ibis/backends/duckdb/__init__.py

```diff
@@ -184,6 +186,44 @@ def register(
 
         return self.table(table_name)
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+    ) -> Iterator[pa.RecordBatch]:
+        pa = self._import_pyarrow()
+        query_ast = self.compiler.to_ast_ensure_limit(
+            expr, limit, params=params
+        )
+        sql = query_ast.compile()
+
+        cursor = self.raw_sql(sql)
+
+        _reader = cursor.cursor.fetch_record_batch(chunk_size=chunk_size)
+        # Horrible hack to make sure cursor isn't garbage collected before batches
```

Comment:
Was just about to!  And I was going to ping you, because I can't tell if DuckDB is doing the right thing and duckdb_engine should copy, or vice-versa, or if there is no right thing at all